app/controllers/products_controller.rb

```diff
@@ -0,0 +1,45 @@
+class ProductsController < ApplicationController
+
+  def new
+    @vendor_id = params[:vendor_id]
+    @product = Product.new
+
+    render :new
+  end
+
+  def create
+    Product.create(create_product_params[:product])
+
+    redirect_to controller: "vendors", action: "products", id: params[:product][:vendor_id]
+  end
+
+  def edit
+    @product = Product.find(params[:product_id])
+
+    render :edit
+  end
+
+  def update
+    @product = Product.find(params[:product][:id])
+    @product.update(create_product_params[:product])
+
+    redirect_to controller: "vendors", action: "products", id: params[:product][:vendor_id]
+  end
+
+  def destroy
+    # we will first destroy all sales assoc. with this product
+    @sales = Sale.where(product_id: params[:product_id])
+    @sales.destroy_all
+
+    vendor_id = params[:id]
+    Product.destroy(params[:product_id])
+
+    redirect_to controller: "vendors", action: "products", id: vendor_id
+  end
+
+  private
+
+  def create_product_params
```

Comment:
This works but you can also call it just `product_params` and have a shorter method name that will easily work for both create and update.